### ISSUE TITLE
Warning: Not Supported (Changing an application's revision and channel at the same time.)

### ISSUE THREAD

**Issue by mcarvalhor** [juju/terraform-provider-juju#498]
### Description

Hi Team,

We should be able to change both revision and channel at the same time using the Juju provider.
When trying to do so currently we get this message:
```
│ Warning: Not Supported
│ 
│   with module.cos-lite.juju_application.cos-ingress,
│   on ../../modules/cos-lite-traefik-ingress/main.tf line 204, in resource "juju_application" "cos-ingress":
│   95: resource "juju_application" "cos-ingress" {
│ 
│ Changing an application's revision and channel at the same time.
```

This is supported by Juju CLI, and it's currently the workaround. For example, this works:
`juju refresh --channel latest/stable --revision 191 cos-ingress`

### Urgency

Casually reporting

### Terraform Juju Provider version

0.10.1

### Terraform version

v1.8.5

### Juju version

3.1.8

### Terraform Configuration(s)

```terraform
resource "juju_application" "traefik" {
  name  = "cos-ingress"
  model = "my-juju-model-name"

  charm {
    name     = "traefik-k8s"
    channel  = "latest/edge"   # Then change to "latest/stable" and try to apply.
    revision = 177             # Then change to 191 and try to apply.
  }

  config = {
    juju-external-hostname = "my-website.canonical.com"
    external_hostname      = "my-website.canonical.com"
  }

  expose {}
  units = 1
  trust = true
}
```

### Reproduce / Test

```shell
$ terraform init && terraform apply -auto-approve
# Applied successfully.

$ # Now change *.tf file: channel to "latest/stable" and revision to 191.

$ terraform init && terraform apply -auto-approve
# Fails.
```

**Comment by hmlanigan** [comment]
I don't believe that juju should allow this. It many ways its non-deterministic behavior. 

@mcarvalhor What is the use case for this operation?

**Comment by mcarvalhor** [comment]
Hi @hmlanigan !

Juju CLI already allows this. For instance, I can run this without any issues:
`juju refresh --channel latest/stable --revision 191 cos-ingress`

The use case is: just the need to change the charm channel.
(for example, from "latest/stable" to "latest/edge")

When we change any charm channel, that consequently means we also need to change the revision, because a revision is tied to a specific channel. For this reason, I believe this error doesn't make sense (we can't change a channel without changing a revision too, and even Juju CLI allows that).

**Comment by mcarvalhor** [comment]
**Workaround:** use Juju CLI instead of terraform to apply the change (using the command I pasted as example above ^).

**Comment by yanksyoon** [comment]
@hmlanigan - just out of curiosity, may I ask what the non-deterministic behavior might be? Would this be on the Juju side or the charm side?

**Comment by beliaev-maksim** [comment]
@hmlanigan If we use revision to be deployed, but channel to track for future upgrades, where does the conflict appear?
That is how CLI works. Currently, we are enforced to run TF twice for this simple operation.

**Comment by luci1900** [comment]
The restriction has been lifted in this PR https://github.com/juju/terraform-provider-juju/pull/889.
